modules/core/shared/src/main/scala/org/legogroup/woof/EnclosingClass.scala

```diff
@@ -0,0 +1,15 @@
+package org.legogroup.woof
+
+case class EnclosingClass(fullName: String, lineLength: Int = 80):
+  lazy val printableName: String = reduceNameLength(fullName)
+
+  private def reduceNameLength(name: String): String =
+    List
+      .unfold(name.split('.').toList) {
+        case Nil                                                        => None
+        case last :: Nil                                                => Some(last, Nil)
+        case parts @ x :: xs if parts.mkString(".").length > lineLength => Some(x.take(1), xs)
```

Comment:
I realize that there's a subtle bug here. We're checking the length of the remaining parts only, whereas it should probably be the "chopped down parts" + the remaining parts. In reality, this implementation is fine since no-one is forcing this to be a strict line length, and the result will be "close enough".
I'll give it some thought to see if we can fix this, otherwise we'll keep it as is.